app/signup/_components/SignUpForm.tsx

```diff
@@ -0,0 +1,50 @@
+"use client";
+
+import { redirect } from "next/navigation";
+import { useEffect } from "react";
+import { useFormState } from "react-dom";
+import { signup } from "~/actions/auth";
+import { Button } from "~/components/ui/button";
+import { Input } from "~/components/ui/input";
+import { Label } from "~/components/ui/label";
+
+const SignUpForm = () => {
+  const initialState = { error: null, success: false };
+  const [formState, formAction] = useFormState(signup, initialState);
+
+  useEffect(() => {
```

Comment:
I think a better way to handle redirect would be to:
1. create a utility function to check for session
2. Use it to redirect at the page level once there is a session

I think this is the approach we take in Fresco. 

Also, the signup action logs the user in, so theoretically we would want this to redirect to `/`. There's no need for the user to log in again. Implementing in the suggested way above should take care of this though.

Reply:
Resolved with this https://github.com/complexdatacollective/Studio/pull/13/commits/b3093549a6c4e812085a3b438f9462ab4397d5cc